[IMP] inventory: flesh out accounting

Felicious opened the pull request:
Revision of [Average inventory valuation cost method](https://www.odoo.com/documentation/14.0/applications/finance/accounting/others/inventory/avg_price_valuation.html).
Task ID: #[3255051](https://www.odoo.com/web#id=3255051&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

### Summary
Hi! I'm newly assigned to the inventory scope and wanted to update this inventory valuation doc for Odoo 14 and elaborate the accounting concepts in layman's terms. If you are able to fact-check and expand on the accounting concepts, I would greatly appreciate it!

**Other changes**
- Add math formulas to elaborate the journal entry calculations about inventory valuation
- Add more headings for organization (because the document got LONG very quickly)
- Renamed headings to be action-oriented and sentence case
- Add external links to related docs
- Remove instances of "I", "our", "we" and other pronouns
- Rewritten for clarity (not brevity)

### Things to look out for in the review
(in order of importance)
1. What do you think about the Anglo-Saxon section (lines 186-266)
2. Does the definition of AVCO make sense?
3. Are the accounting concepts clear & accurate?
4. Do math explanations make sense?
5. Can they be shortened a little?
6. How is the organization of examples?
7. Do intros of sections seem redundant?

This PR is intended to be FW ported

toaa-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,266 @@
+
+.. _inventory/avg_cost/definition:
+
+Average cost valuation (AVCO) is a method of accounting for a company's on-hand stock to reflect the
+value of a company's assets. This method evaluates a product based on its average cost, divided by
+the amount of stock on-hand. It is also *automatically updated* each time products are received.
```
It's missing a little bit of information: it evaluates a cost based on the total costs of goods bought or produced during a period, divided by the total number of items on hand.

You can find more information here: https://www.investopedia.com/terms/a/averagecostmethod.asp

toaa-odoo reviewed (commented):
Hi Felicia! Thanks for your tremendous work, this page clearly needed a bit of love. I have yet to investigate the Anglo-Saxon section and see about that $2 difference, but here's what I have to say about the rest of the PR so far :). I'll get back to you when I know better about the right behavior

Felicious commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,266 @@
+Returning to the previous transactions involving tables, the chart below reflects journal entries
+and accounts. The **stock input** account is meant to store the money intended to pay vendors when
+the vendor bill has yet been received. There is an additional **price difference** account that
+records the difference between price the product is valuated at and the price the vendor sold the
```
Good suggestion! There's a lot happening, so these distinctions are very helpful.

Felicious commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,266 @@
+
+.. _inventory/avg_price/leaving_inventory:
+
+Update cost when products leave: the consequences
```
HAHA! Let me think and come up with another (:

Felicious commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,266 @@
+| 3. Deliver 10 tables           | -10*$12       | $24               | 2             | $12        |
++--------------------------------+---------------+-------------------+---------------+------------+
+
+To start, there are 0 tables in stock, so all values are $0.
```
Noted, @toaa-odoo ! Moved this line below the heading.

Felicious commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,266 @@
+.. math::
+   Avg~Cost = \frac{Old~Qty \times Old~Avg~Cost + Incoming~Qty \times Purchase~Price}{Final~Qty}
+
+- **Old Qty**: count of the product in stock the moment before receiving the new shipment.
```
Thanks for the suggestion! I also prefer brevity.

Felicious commented:
Hi @toaa-odoo ! Thank you again for a thorough review of this doc. I implemented some changes you suggested.

Also, since I'm still familiarizing myself with Github, I accidentally removed you as a contributor in the commit message when I was merging the remote branch with my local branch. So, the commit currently isn't showing that you contributed to the doc. (But the changes you suggested are still there) I will work on restoring it another time!

toaa-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,477 @@
+
+#. Return 1 table to the vendor who sold it at `$10`
+>>>>>>> 9363ec06a ([IMP]: flesh out accounting and math)
+   #. Debit *stock input* account `$12` (Why?)
+      #. also how to account for the `$2` price difference?
```
The *Stock Input* account is debited $12 because the avg cost of a table is 12 at the moment the product is returned (even though it was bought for $10). The $2 price difference has to be accounted for in the *Price Difference Account* field under the product category. The *Price Difference Account* differs from localization to localization of course :).

Note that this only works with automated inventory valuation, and it could be interesting to add as note (+ :doc: or :ref: to this page https://www.odoo.com/documentation/16.0/applications/inventory_and_mrp/inventory/management/reporting/inventory_valuation_config.html?highlight=inventory%20valuation)

toaa-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,477 @@
+
+.. _inventory/avg_cost/definition:
+
+<<<<<<< HEAD
```
```suggestion
<<<<<<< HEAD
```
There are some issues with the multiple commits made. No worries, this is part of the learning process. I'll let you see with your GitHub/PyCharm guru how to fix those. You can ping me again for the final green light once everything's fixed :)

toaa-odoo reviewed (changes requested):
Thanks for the changes, see my new comments to know what's left to fix! Thanks again for your work

Felicious commented:
### Summary of recent edits
4/17/23

1. Added images of Odoo screenshots to show workflow
2. Added configuration instructions and descriptions for where to find buttons/menus to use AVCO in Odoo

Concerns:
- the length
-   In what ways can I better use admonition blocks, cards, and other RST tools to help users find the information they need and sift through this long doc?
- Thinking about: the flow and presentation of information. ie. Should the theoretical information be hidden using **spoiler** blocks? Are the charts useful? How about the summary of debit/credit statements?

toaa-odoo reviewed (changes requested):
Hi @Felicious, thanks for the improvement of the documentation. I made a few notes and comments, but I'll try to tackle the pain points you mentioned.

1) Length: as of right now, the doc is rather long indeed. I feel the previous commit had the perfect length, but it did not contain any "Odoo flow", which may or may not be necessary to include.
2) RST tools: the admonition blocks etc, are fine (IMO), however, one way to tackle your 1st and 3rd point could be to use "tabs". My idea is, keeping the current structure (with a few tweaks), you could have the H-heading "Stock products for the first time", and then have two tabs. The first one contains all the theory (and would be open by default and be the first information the user sees). The second tab (let's call it "Odoo flow"), would contain all the flow from A to Z (related to that H-heading), just like what is currently contained in the .. example:: admonition block. That way, it would effectively make the page less long and also structure the info in two separate places. One is the theory, and the other is the flow (a bit like an "optional" or "extra info" space). The doc would thus contain all the theory by default, and when applicable (or useful), have an "Odoo flow" tab where users could see exactly how it works in Odoo if they wish

This is just an idea, I do not know what it would look like or work, but I feel it could maybe tackle the aforementioned pain points. I'll let you experiment :)

meng-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,479 @@
+    :align: center
+    :alt: Accounting entry for receiving 4 tables
+
+#. Like #2, when the bill for 4 tables is received,
```
Same here, unsure what #2 refers to.

meng-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,479 @@
+   :align: center
+   :alt: Show journal items linked to sale order
+
+.. spoiler:: Understand Anglo-Saxon expensing (that's slightly unrelated to inventory valuation)
+   In the image above, the accounts **Product Sales**, **Tax Received**, and **Accounts Receivable**
+   all pertain to the sale of the product. **Accounts Receivable** is the account where the customer
+   payment will be received.
```
The formatting for this box didn't come through correctly in the html version.

meng-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,479 @@
+   Behavior of *price difference accounts* varies from localization. In this case, the account is
+   intended to store differences between vendor price and *automated* inventory valuation methods.
+
+A summary of the image below:
```
Don't reference the image directly :)

meng-odoo reviewed (changes requested):
Hi @Felicious! Great job on this doc, it's super thorough and well-written! I made suggestions for formatting, content, and wording throughout, which I hope are helpful in shortening the doc and making it a bit more skimmable and understandable for readers. Please let me know if you have any questions!

meng-odoo reviewed (changes requested):
Hi @Felicious! Great job on this doc, it's super thorough and well-written! I made suggestions for formatting, content, and wording throughout, which I hope are helpful in shortening the doc and making it a bit more skimmable and understandable for readers. Please let me know if you have any questions!

Felicious commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,479 @@
+
+Odoo generates an accounting journal entry when shipments that use the :abbr:`AVCO (Average Cost
+Valuation)` costing method are received. Configure a :guilabel:`Price Difference Account` by
+selecting the :guilabel:`-->` icon next to the :guilabel:`Product Category` field on the product
```
```suggestion
selecting the :guilabel:`➡️ (arrow)` icon next to the :guilabel:`Product Category` field on the product
```

meng-odoo commented on content/applications/finance/accounting/others/inventory/avg_price_valuation.rst:
```diff
@@ -1,167 +1,442 @@
+.. example::
+
+   To return a single table that was purchased for `$10`, navigate to the receipt in the *Inventory*
+   app for the :ref:`8 tables purchased in Exercise 1 <inventory/avg_cost/ex-1>`.
```
I believe this is how you get there, but please correct this if I'm wrong
```suggestion
   app for the :ref:`8 tables purchased in Exercise 1 <inventory/avg_cost/ex-1>` by going to the :guilabel:`Inventory Overview` and clicking on :guilabel:`Receipts`.
```

meng-odoo reviewed (approved):
Hi @Felicious! This doc is very well written, concise, and easy to follow, and I appreciate you including many links to elsewhere in the doc or to other docs, which makes it super usable for readers skimming :) I had some small additions to make some of the instructions clearer/more detailed, and small typo/formatting fixes. I'm approving now, but please go through those edits before moving this to the next review stage. Thank you!

Felicious commented:
Thank you, @meng-odoo :smile:, really appreciate your thorough review! I'm very confident in this doc, now. Passing it over to you, @StraubCreative

StraubCreative reviewed (approved):
LGTM, all checks pass ✅ 

Since this has been open for a bit, I'd like to ship live and we can address any changes in a new PR.

Thanks everyone for the detailed work on this!

@robodoo r+
